### PR TITLE
Addons updates

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:
Update some addons. For others add instructions to migrate away from addons to helmReleases.

**Which issue(s) this PR fixes**:

Ref #2562

**What type of PR is this?**
/kind chore

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
Update addons:
* backup-restic
* unattended-updates
* csi-vault-secret-provider
* secrets-store-csi-driver
```

**Documentation**:
```documentation
NONE
```

### CODE REVIEW

File: addons/csi-vault-secret-provider/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# Vault CSI driver
+# !!!WARNING!!! USING THIS ADDON IS NO RECOMMEDED ANYMORE, PLEASE MIGRATE TO HELM RELEASE! Vault CSI driver
```
Comment (xmudrii) [inline addons/csi-vault-secret-provider/README.md:1]:
```suggestion
# !!!WARNING!!! USING THIS ADDON IS NOT RECOMMENDED ANYMORE, PLEASE MIGRATE TO HELM RELEASE! Vault CSI driver
```

File: addons/secrets-store-csi-driver/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Secrets Store CSI Driver
+# !!!WARNING!!! USING THIS ADDON IS NO RECOMMEDED ANYMORE, PLEASE MIGRATE TO HELM RELEASE! Secrets Store CSI Driver
```
Comment (xmudrii) [inline addons/secrets-store-csi-driver/README.md:1]:
```suggestion
# !!!WARNING!!! USING THIS ADDON IS NOT RECOMMENDED ANYMORE, PLEASE MIGRATE TO HELM RELEASE! Secrets Store CSI Driver
```

File: addons/csi-vault-secret-provider/README.md (hunk 1 above)
Comment (kron4eg) [reply addons/csi-vault-secret-provider/README.md]: :+1:

File: addons/secrets-store-csi-driver/README.md (hunk 2 above)
Comment (kron4eg) [reply addons/secrets-store-csi-driver/README.md]: :+1:

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**